viewing_party/party.py

```diff
@@ -0,0 +1,344 @@
+                if movie not in friends_watched:
+                    friends_watched.append(movie)
+
+        for movie in user_watched:
+            if movie not in friends_watched:
+                user_unique_watched.append(movie)
+
+        return user_unique_watched
+    return []
+        
+    # *_____________________________ with many lists =S  __________________________________* 
+    # list_of_friends=user_data["friends"]
+    # list_of_movies=[]
+    # list_of_titles=[]
+    # friends=[]
+    # amigo_movie=[]
+    # friends_watched=[]
+    # for friend in list_of_friends:
+    #     list_of_movies.append(friend["watched"])
+    # for friend in list_of_friends:
+    #     friends.append(friend)
+    # for friend in list_of_movies:
+    #     list_of_titles.append(friend)
+    # for objeto in list_of_titles:
+    #     for amigo in objeto:
+    #         amigo_movie.append(amigo)
+    # for amigo in amigo_movie:
+    #     friends_watched.append(amigo["title"])
+    # # print(friends) 
+    # # print(list_of_friends) 
+    # # print(list_of_titles)
+    # # print(amigo_movie)
+    # # print(friends_watched)
+    # user_movies=user_data["watched"]
+    # user_watched=[]
+    # for movie in user_movies:
+    #     user_watched.append(movie["title"])
+ 
+    # user_set=set(user_watched)
+    # friends_set=set(friends_watched)
+    # result_set=user_set.difference(friends_set)
+    
+    # final_list=[]
+    # for title in result_set:
+    #     final_dict={}
+    #     final_dict["title"]=title
+    #     # print(title)
+    #     # print(final_dict)
+    #     final_list.append(final_dict)
+    
+    # return final_list
+
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+    
+def get_friends_unique_watched(user_data):
+    list_of_friends=user_data["friends"]
+    list_of_movies=[]
+    list_of_titles=[]
+    friends=[]
+    amigo_movie=[]
+    friends_watched=[]
+    for friend in list_of_friends:
+        list_of_movies.append(friend["watched"])
+    for friend in list_of_friends:
+        friends.append(friend)
+    for friend in list_of_movies:
+        list_of_titles.append(friend)
+    for objeto in list_of_titles:
+        for amigo in objeto:
+            amigo_movie.append(amigo)
+    for amigo in amigo_movie:
+        friends_watched.append(amigo["title"])
+
+    user_movies=user_data["watched"]
+    user_watched=[]
+    for movie in user_movies:
+        user_watched.append(movie["title"])
+ 
+    user_set=set(user_watched)
+    friends_set=set(friends_watched)
+    result_set=friends_set.difference(user_set)
+    
+    final_list=[]
+    for title in result_set:
+        final_dict={}
+        final_dict["title"]=title
```

Comment:
This will pass our tests, but what if the movie dictionaries had ratings or genres?  This step would return a list of dictionaries that only contain the titles.  How could you use each title in the set of titles to find the original dictionary that it came from?